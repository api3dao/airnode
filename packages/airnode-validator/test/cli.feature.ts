import { spawnSync } from 'child_process';
import { join } from 'path';

const runValidator = (args: string[]) => {
  const command = ['node', join(__dirname, '../dist/bin/validator.js'), ...args].join(' ');

  return spawnSync(command, { shell: true });
};

describe('validator CLI', () => {
  it('shows help', () => {
    const cliHelp = runValidator(['--help']).stdout.toString();

    expect(cliHelp).toMatchSnapshot();
  });

  it('validates valid configuration', () => {
    const args = [
      `--config ${join(__dirname, './fixtures/config.valid.json')}`,
      `--secrets ${join(__dirname, './fixtures/secrets.valid.env')}`,
    ];

    const output = runValidator(args);

    expect(output.status).toBe(0);
    // We use "expect.stringContaining" because the output begins with "✔"
    expect(output.stderr.toString()).toEqual(expect.stringContaining('The configuration is valid\n'));
  });

  it('validates invalid configuration', () => {
    const args = [
      `--config ${join(__dirname, './fixtures/config.valid.json')}`,
      `--secrets ${join(__dirname, './fixtures/missing-secrets.env')}`,
    ];

    const output = runValidator(args);

    expect(output.status).toBe(1);
    expect(output.stderr.toString()).toEqual(
      // We use "expect.stringContaining" because the output begins with "✖"
      expect.stringContaining(
        'The configuration is not valid. Reason: Secrets interpolation failed. Caused by: PROVIDER_URL is not defined'
      )
    );
  });
});
