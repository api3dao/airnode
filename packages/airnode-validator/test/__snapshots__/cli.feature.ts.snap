// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validator CLI shows help 1`] = `
"Options:
      --version  Show version number                                                                           [boolean]
  -c, --config   Path to \\"config.json\\" file to validate                                              [string] [required]
  -s, --secrets  Path to \\"secrets.env\\" file to interpolate in the config                             [string] [required]
      --help     Show help                                                                                     [boolean]

Examples:
  --config pathTo/config.json --secrets pathTo/secrets.env
  -c pathTo/config.json -s pathTo/secrets.env
"
`;
